harness/engine/utils.go

```diff
@@ -481,3 +527,84 @@ func (n *Network) SendTransferTxCmd(senderNodeInd, recvNodeInd uint, amount, fee
 
 	return resp.Hash, nil
 }
+
+// BatchSendTransferTx sends a transfer call from node of index senderNodeInd to senderNodeInd+1 node.
+func (n *Network) BatchSendTransferTx(t *testing.T, senderNodeInd uint, batchSize uint, amount, fee uint64, timeout time.Duration) error {
+	recvNodeInd := senderNodeInd + 1
+	if recvNodeInd >= uint(n.Size()) {
+		recvNodeInd = 0
+	}
+
+	// Get wallet address of sender
+	senderAddr, _, err := n.GetWalletAddress(senderNodeInd)
+	if err != nil {
+		return err
+	}
+
+	// Get wallet address of receiver
+	recvAddr, pubKey, err := n.GetWalletAddress(recvNodeInd)
+	if err != nil {
+		return err
+	}
+
+	if logrus.GetLevel() == logrus.DebugLevel {
+		logrus.
+			WithField("s_wallet", senderAddr).
+			WithField("r_wallet", recvAddr).
+			WithField("amount", amount).
+			WithField("fee", fee).
+			Debug("Sending transfer")
+	}
+
+	// Send Transfer grpc command
+	c := n.grpcClients[n.nodes[senderNodeInd].Id]
+
+	conn, err := c.GetSessionConn(grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		_ = conn.Close()
+	}()
+
+	client := pb.NewTransactorClient(conn)
+
+	for i := uint(0); i < batchSize; i++ {
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+
+		req := pb.TransferRequest{Amount: amount, Address: pubKey, Fee: fee}
+
+		_, err := client.Transfer(ctx, &req)
+		if err != nil {
+			logrus.WithField("sender_index", senderNodeInd).Error(err)
+		}
+	}
+
+	return nil
+}
+
+// MonitorTPS tries to measure network TPS each N seconds.
+func (n *Network) MonitorTPS(delay time.Duration) {
+	// Start monitoring TPS value of the network
+	nodeInd := uint(0)
+
+	for {
+		time.Sleep(delay)
+
+		h, tps, blockTime, lastTxsCount, err := n.CalculateTPS(nodeInd)
+
+		if err == nil {
+			logrus.WithField("height", h).Infof("Measured TPS %.2f (%d/%d)", tps, lastTxsCount, blockTime)
+		} else {
```

Comment:
In accordance with the coding conventions, let's try and avoid using `else` whenever possible; this is because we try to avoid extra indentations where it's not needed.

You could rewrite it as such:

```golang
if err != nil {
    logrus.WithError(err).Warn("calculate tps failed")

    // Failover to another node if this fails
    nodeInd++
    if nodeInd >= uint(n.Size()) {
        nodeInd = 0
    }

    continue
}

logrus.WithField("height", h).Infof("Measured TPS %.2f (%d/%d)", tps, lastTxsCount, blockTime)
```